mbq/client/client.py

```diff
@@ -27,6 +28,14 @@ def _make_url(self, url):
     def _make_request(self, method, url, *args, **kwargs):
         if self._headers:
             kwargs['headers'] = dict(self._headers, **kwargs.get('headers', {}))
+        else:
+            kwargs['headers'] = {}
+
+        if 'correlation_id' in kwargs:
```

Comment:
Spitballing:

imagine the following workflow is using mbq.client for each hop
BFF -> CX -> Darkside -> Iris
this should generate (at least) 4 SNS workflow events (one from each service), each of which would have the same correlation id.

I am wondering if there is a way to instrument automatically storing received correlation-ids from the incoming request and then magically have detected and forwarded with outgoing requests?

Reply:
wondering the same thing. Let's keep our eyes open for such an opportunity.

Reply:
Might be misunderstanding Ryan's question but I've used thread locals and global middleware before to make this process relatively magical. Easy to consume those thread locals in logs, outbound requests, etc

Reply:
agree. I think that is Ryan's question. `mbq.client` is agnostic of the web framework so I don't think that it's the right place to put that stuff.